### PR TITLE
Internal: Checklist will be visible for users with Admin capabilities [ED-15633]

### PR DESCRIPTION
**Pull request by elchugreeva**

### CODE REVIEW

File: tests/playwright/sanity/modules/checklist/checklist.test.ts [hunk 1]
```diff
@@ -333,4 +341,27 @@ test.describe( 'Launchpad checklist tests', () => {
 
 		await checklistHelper.toggleExpandChecklist( 'editor', true );
 	} );
+	test( 'Checklist visible only to admin', async ( { browser, page, apiRequests }, testInfo ) => {
+		await test.step( 'Checklist visible to admin default', async () => {
```
Comment (davseve) [inline tests/playwright/sanity/modules/checklist/checklist.test.ts:345]:
```suggestion
		await test.step( 'Checklist visible to admin role by default', async () => {
```

File: tests/playwright/sanity/modules/checklist/checklist.test.ts [hunk 2]
```diff
@@ -333,4 +341,27 @@ test.describe( 'Launchpad checklist tests', () => {
...
+			const wpAdmin = new WpAdminPage( page, testInfo, apiRequests ),
+				editor = await wpAdmin.openNewPage(),
+				rocketButton = editor.page.locator( selectors.topBarIcon ),
+				checklist = editor.page.locator( selectors.popup );
+			await rocketButton.click();
+			await expect( checklist ).toBeVisible();
+		} );
+
+		await test.step( 'Checklist not visible to author', async () => {
```
Comment (davseve) [inline tests/playwright/sanity/modules/checklist/checklist.test.ts:354]:
```suggestion
		await test.step( 'Checklist not visible to author role', async () => {
```

File: modules/checklist/module.php [hunk 3]
```diff
@@ -54,6 +54,11 @@ public function __construct( ?Wordpress_Adapter_Interface $wordpress_adapter = n
 		Plugin::$instance->data_manager_v2->register_controller( new Controller() );
 		$this->user_progress = $this->user_progress ?? $this->get_user_progress_from_db();
 		$this->steps_manager = new Steps_Manager( $this );
+
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return;
+		}
```
[the remark this answers is not in the document]
Comment (elchugreeva) [reply modules/checklist/module.php:60]: @davseve it breaks all unit tests if it's on top

File: tests/playwright/sanity/modules/checklist/checklist.test.ts (hunk 1 above)
Comment (elchugreeva) [reply tests/playwright/sanity/modules/checklist/checklist.test.ts]: fixed

File: tests/playwright/sanity/modules/checklist/checklist.test.ts (hunk 2 above)
Comment (elchugreeva) [reply tests/playwright/sanity/modules/checklist/checklist.test.ts]: fixed

File: tests/playwright/assets/api-requests.ts [hunk 4]
```diff
@@ -234,4 +258,36 @@ export default class ApiRequests {
 
 		return response;
 	}
+
+	public async createNewUser( request: APIRequestContext, user ) {
```
Comment (ronenelementor) [inline tests/playwright/assets/api-requests.ts:262]: Assuming there is a `User` type, if not, I think we should create one
```suggestion
	public async createNewUser( request: APIRequestContext, user: User ) {
```

**Review by ronenboxer — APPROVED** [review]

**Review by davseve — APPROVED** [review]

**Pull request merged**